hkmc2/shared/src/test/mlscript-compile/LazyArray.mls

```diff
@@ -0,0 +1,137 @@
+
+import "./Iter.mls"
+
+open Iter
+open LazyArray
+
+fun normIdx(i, len) =
+  if i < 0 then len + i
+  else i
+
+// Assume all underlying `IterableBase`s are constant-time random access
```

Comment:
This is a very confusing comment. We shouldn't make any such assumptions. Are you trying to say what the contract of `LazyArr` should be? Or is it a contract of the `IterableBase` class?

Reply:
This is rather poorly worded - It's a contract of `LazyArr` and should be made clear to users of lazy spreads. 

I'll rephrase to make it sound less like the latter.